Mergin/plugin.py

```diff
@@ -575,71 +625,116 @@ def __init__(self, parent, grp_name, grp_filter, icon, order, project_manager):
         self.setIcon(QIcon(icon_path(icon)))
         self.setSortKey(order)
         self.project_manager = project_manager
-
-    def createChildren(self):
-        mc = self.parent().mc
-        if not mc:
+        self.projects = []
+        self.group_name = grp_name
+        self.total_projects_count = None
+        self.fetch_more_item = None
+
+    def fetch_projects(self, page=1, per_page=PROJS_PER_PAGE):
+        """Get paginated projects list from Mergin service and an error message list."""
+        if self.project_manager is None:
             error_item = QgsErrorItem(self, "Failed to login please check the configuration", "/Mergin/error")
             sip.transferto(error_item, self)
             return [error_item]
         try:
-            projects = mc.projects_list(flag=self.filter)
+            resp = self.project_manager.mc.paginated_projects_list(
+                flag=self.filter, page=page, per_page=per_page, order_by="namespace")
+            self.projects += resp["projects"]
+            self.total_projects_count = int(resp["count"]) if is_number(resp["count"]) else 0
         except URLError:
             error_item = QgsErrorItem(self, "Failed to get projects from server", "/Mergin/error")
             sip.transferto(error_item, self)
             return [error_item]
         except Exception as err:
             error_item = QgsErrorItem(self, "Error: {}".format(str(err)), "/Mergin/error")
             sip.transferto(error_item, self)
-            return [error_item]
+            return None, [error_item]
+        return None
```

Comment:
Hmm what happens with the return values? in different occasions we return `[error_item]` or `None, [error_item]` or `None` ? What are we supposed to return?

Reply:
Yep, this is a mess. We either need None (all OK) of an error item (as a list item), if something goes wrong.